alot/__main__.py

```diff
@@ -13,115 +14,67 @@
-        print alot.__version__
-        sys.exit(0)
-
-
-class ComposeOptions(SubcommandOptions):
-    optParameters = [
-        ['sender', '', None, 'From line'],
-        ['subject', '', None, 'subject line'],
-        ['to', [], None, 'recipients'],
-        ['cc', '', None, 'copy to'],
-        ['bcc', '', None, 'blind copy to'],
-        ['template', '', None, 'path to template file'],
-        ['attach', '', None, 'files to attach'],
-    ]
-    optFlags = [
-        ['omit_signature', '', 'do not add signature'],
-    ]
-
-    def parseArgs(self, *args):
-        SubcommandOptions.parseArgs(self, *args)
-        self.rest = ' '.join(args) or None
-
-
-class SearchOptions(SubcommandOptions):
-    accepted = ['oldest_first', 'newest_first', 'message_id', 'unsorted']
-
-    def colourint(val):
-        if val not in accepted:
-            raise ValueError("Unknown sort order")
-        return val
-    colourint.coerceDoc = "Must be one of " + str(accepted)
-    optParameters = [
-        ['sort', 'newest_first', None, 'Sort order'],
-    ]
-
-
-class Options(usage.Options):
-    optFlags = [["read-only", "r", 'open db in read only mode'], ]
-
-    def colourint(val):
-        val = int(val)
-        if val not in [1, 16, 256]:
-            raise ValueError("Not in range")
-        return val
-    colourint.coerceDoc = "Must be 1, 16 or 256"
-
-    def debuglogstring(val):
-        if val not in ['error', 'debug', 'info', 'warning']:
-            raise ValueError("Not in range")
-        return val
-    debuglogstring.coerceDoc = "Must be one of debug,info,warning or error"
-
-    optParameters = [
-        ['config', 'c', None, 'config file'],
-        ['notmuch-config', 'n', None, 'notmuch config'],
-        ['colour-mode', 'C', None, 'terminal colour mode', colourint],
-        ['mailindex-path', 'p', None, 'path to notmuch index'],
-        ['debug-level', 'd', 'info', 'debug log', debuglogstring],
-        ['logfile', 'l', '/dev/null', 'logfile'],
-    ]
-    search_help = "start in a search buffer using the querystring provided "\
-                  "as parameter. See the SEARCH SYNTAX section of notmuch(1)."
-
-    subCommands = [['search', None, SearchOptions, search_help],
-                   ['compose', None, ComposeOptions, "compose a new message"]]
-
-    def opt_version(self):
-        print alot.__version__
-        sys.exit(0)
-
 
 def main():
-    # interpret cml arguments
-    args = Options()
-    try:
-        args.parseOptions()  # When given no argument, parses sys.argv[1:]
-    except usage.UsageError as errortext:
-        print '%s' % errortext
-        print 'Try --help for usage details.'
-        sys.exit(1)
+    # set up the parser to parse the command line options.
+    parser = argparse.ArgumentParser(
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+    parser.add_argument('-v', '--version', action='version',
+                        version='%(prog)s, {}'.format(alot.__version__))
```

Comment:
my bad, will change it if nobody comes forth and demands the new output to stay.